internal/storage/gcs/folder.go

```diff
@@ -15,11 +15,34 @@
 package gcs
 
 import (
+	"strings"
 	"time"
+
+	"cloud.google.com/go/storage/control/apiv2/controlpb"
 )
 
 type Folder struct {
 	Name           string
 	Metageneration int64
 	UpdateTime     time.Time
 }
+
+func ConvertToGCSFolderFromGetFolderResponse(bucketName string, attrs *controlpb.Folder) *Folder {
```

Comment:
nit Could we call this getGCSFolder() instead of a elaborateName.

Reply:
It is converting the type, that's why I feel convert make sense here.

Reply:
as per go/go-style/best-practices#function-name-conventions it says even noun names should be good. i.e. gcsFolder(bucketName, attrs *controlpb.Folder) should be fine.
But if this is how the naming code is already present in other files in our codebase, its fine to do it later. Will let you take the call for it.

Reply:
Ohh okay.. then let's do it.

Reply:
Done